osgp/protocol-adapter-oslp/osgp-adapter-protocol-oslp-elster/src/main/java/org/opensmartgridplatform/adapter/protocol/oslp/elster/application/config/OslpConfig.java

```diff
@@ -70,74 +60,79 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(OslpConfig.class);
 
-    @Bean(destroyMethod = "releaseExternalResources")
-    public ClientBootstrap clientBootstrap() {
-        InternalLoggerFactory.setDefaultFactory(new Slf4JLoggerFactory());
-        final ChannelFactory factory = new NioClientSocketChannelFactory(Executors.newCachedThreadPool(),
-                Executors.newCachedThreadPool());
+    @Bean(name = "protocolAdapterOslpNettyServerBossGroup")
+    public DisposableNioEventLoopGroup serverBossGroup() {
+        // TODO - Should we pass in any parameters, like for example number of
+        // threads, to the constructor?
+        return new DisposableNioEventLoopGroup();
+    }
 
-        final ChannelPipelineFactory pipelineFactory = new ChannelPipelineFactory() {
-            @Override
-            public ChannelPipeline getPipeline() throws ProtocolAdapterException {
-                final ChannelPipeline pipeline = OslpConfig.this
-                        .createChannelPipeline(OslpConfig.this.oslpChannelHandlerClient());
+    @Bean(name = "protocolAdapterOslpNettyServerWorkerGroup")
+    public DisposableNioEventLoopGroup serverWorkerGroup() {
+        // TODO - Should we pass in any parameters, like for example number of
+        // threads, to the constructor?
+        return new DisposableNioEventLoopGroup();
+    }
 
-                LOGGER.info("Created client new pipeline");
+    @Bean(name = "protocolAdapterOslpNettyClientWorkerGroup")
+    public DisposableNioEventLoopGroup clientWorkerGroup() {
+        // TODO - Should we pass in any parameters, like for example number of
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)